private/model/api/passes.go

```diff
@@ -244,6 +244,46 @@ func (a *API) renameExportable() {
 	}
 }
 
+// renameCollidingFields will rename any fields that uses an SDK or Golang
+// specific name.
+func (a *API) renameCollidingFields() {
+	for _, v := range a.Shapes {
+		namesWithSet := map[string]struct{}{}
+		for k, field := range v.MemberRefs {
+			if strings.HasPrefix(k, "Set") {
+				namesWithSet[k] = struct{}{}
+			}
+
+			if collides(k) {
+				delete(v.MemberRefs, k)
+				v.MemberRefs[k+"_"] = field
+			}
+		}
+
+		// checks if any field names collide with setters.
+		for name := range namesWithSet {
+			setterName := fmt.Sprintf("Set%s", name)
+			if field, ok := v.MemberRefs[setterName]; ok {
+				delete(v.MemberRefs, name)
+				v.MemberRefs[name+"_"] = field
+			}
+		}
+	}
+
+}
+
+// collides will return true if it is a name used by the SDK or Golang.
+func collides(name string) bool {
+	switch name {
```

Comment:
Nit, this would be nicer as the following. This format puts all names into a single case and easier to read.

```go
func IsGeneratedMethod(name string) bool {
	switch name {
	case
		"String",
		"GoString",
		"Validate":
		return true
	default:
		return false
	}
}
```